### PR TITLE
Imaki dev

### PR DESCRIPTION
**Pull request by vecinomio**
Added Bastion template;
Added diagram to Readme file;
Edited Readme file.

### CODE REVIEW

File: A3-ASG-bastion.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+  AWSTemplateFormatVersion: "2010-09-09"
+
+  Description: "Creates a Launch Configuration with Bastion host and
+  Auto Scaling Group for Bastion in Public subnet 0"
+
+  Parameters:
+    VPCStackName:
+      Type: String
+      Default: DevVPC
```
Comment (mike-unitskyi) [inline A3-ASG-bastion.yaml]: would remove default value

File: A3-ASG-bastion.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      MinLength: 1
```
Comment (mike-unitskyi) [inline A3-ASG-bastion.yaml]: doesn't make sense as `AllowedValues` are specified.

File: A3-ASG-bastion.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      AllowedValues:
+        - DevVPC
+        - ProdVPC
+    KeyName:
+      Description: Name of an existing EC2 KeyPair to enable SSH access to the instance
+      Type: 'AWS::EC2::KeyPair::KeyName'
+      Default: imaki_NVirginia
+      ConstraintDescription: must be the name of an existing EC2 KeyPair.
+    SSHLocation:
```
Comment (mike-unitskyi) [inline A3-ASG-bastion.yaml]: out of curiosity, why this was extracted to parameter?

File: A3-ASG-bastion.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      AllowedValues:
+        - t2.micro
+        - t2.small
+        - t2.medium
+      ConstraintDescription: Must be a valid EC2 instance type.
+
+  Mappings:
+    RegionMap:
+      us-east-1:
+        AMI: ami-0080e4c5bc078760e # Amazon Linux 2
```
Comment (mike-unitskyi) [inline A3-ASG-bastion.yaml]: as VPC stack supports 2 regions it makes sense to add another region here also.

File: A3-ASG-bastion.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
+  AWSTemplateFormatVersion: "2010-09-09"
+
+  Description: "Creates a Launch Configuration with Bastion host and
+  Auto Scaling Group for Bastion in Public subnet 0"
+
+  Parameters:
+    VPCStackName:
```
Comment (mike-unitskyi) [inline A3-ASG-bastion.yaml]: parameter is unused

File: A3-ASG-bastion.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      us-east-1:
+        AMI: ami-0080e4c5bc078760e # Amazon Linux 2
+
+  Resources:
+    BastionSecurityGroup:
+      Type: 'AWS::EC2::SecurityGroup'
+      Properties:
+        GroupDescription: Enable SSH access via port 22
+        VpcId:
+          Fn::ImportValue: "DevVPC-VPCId"
```
Comment (mike-unitskyi) [inline A3-ASG-bastion.yaml]: use `VPCStackName` parameter instead of hardcoded value

File: A3-ASG-bastion.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        KeyName: !Ref KeyName
+        SecurityGroups:
+          - !Ref BastionSecurityGroup
+
+
+    BastionAutoScalingGroup:
+      Type: "AWS::AutoScaling::AutoScalingGroup"
+      Properties:
+        VPCZoneIdentifier:
+          - Fn::ImportValue: "DevVPC-PublicSubnet0"
```
Comment (mike-unitskyi) [inline A3-ASG-bastion.yaml]: use `VPCStackName` parameter instead of hardcoded value

File: A3-ASG-bastion.yaml (hunk 3 above)
Comment (vecinomio) [reply A3-ASG-bastion.yaml]: have found this example in one of the AWS templates

File: A3-ASG-bastion.yaml (hunk 4 above)
Comment (vecinomio) [reply A3-ASG-bastion.yaml]: ok, accepted

File: A3-ASG-bastion.yaml [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    SSHLocation:
+      Description: The IP address range that can be used to SSH to the EC2 instances
+      Type: String
+      Default: 0.0.0.0/0
+      ConstraintDescription: must be a valid IP CIDR range of the form x.x.x.x/x.
+    InstanceType:
+      Type: String
+      Description: Type of EC2 instance.
+      Default: t2.micro
+      AllowedValues:
```
Comment (Emperorex) [inline A3-ASG-bastion.yaml]: I would use shorter record: `AllowedValues: [ t2.micro, t2.small, t2.medium ]`

Comment (vecinomio) [reply A3-ASG-bastion.yaml]: clear, understood.

File: A3-ASG-bastion.yaml (hunk 7 above)
Comment (vecinomio) [reply A3-ASG-bastion.yaml]: understood, will do.

File: A3-ASG-bastion.yaml (hunk 6 above)
Comment (vecinomio) [reply A3-ASG-bastion.yaml]: understood, will do.

File: A3-ASG-bastion.yaml [hunk 9]
```diff
@@ -0,0 +1,76 @@
+  AWSTemplateFormatVersion: "2010-09-09"
+
+  Description: "Creates a Launch Configuration with Bastion host and
+  Auto Scaling Group for Bastion in Public subnet 0"
+
+  Parameters:
+    VPCStackName:
+      Type: String
+      AllowedValues:
```
Comment (IYermakov) [inline A3-ASG-bastion.yaml]: Actually we use this parameter as imported value, so I woudn't set restrictions here

**Pull request closed without merging**